### PR TITLE
pkg/lwip: Add basic version of netif_get/set_opt

### PR DESCRIPTION
**Pull request by yarrick**
### Contribution description

Add basic netif get/set opt implementation for lwIP. 

Only supporting getting IPv6 addresses and underlying netif options. Setting options not supported.

### Testing procedure

Apply e22fe428b4da8f6cc058577ffe0d27b5ed065ced and f502b031f2935e50306fa083e8f9ddebaa98930d, and build and flash `tests/lwip` (l2util module needs to be added).
The gnrc version of `ifconfig` then prints the lwIP netifs including IPv6 addresses, link type and channel for WiFi:

```
> ifconfig 
Iface  ET1  HWaddr: 24:0A:C4:E6:0E:9C  Channel: 11  Link: up 
          L2-PDU:1500  Source address length: 6
          Link type: wireless
          inet6 addr: fe80::260a:c4ff:fee6:e9c  scope: link
          inet6 addr: 2001:db8:0:0:260a:c4ff:fee6:e9c  scope: global
[..]
```

### Issues/PRs references

Part of #9287

### CODE REVIEW

**Review by benpicco — APPROVED** [review]
A surprisingly simple addition, nice to have this now work across stacks.

File: pkg/lwip/contrib/_netif.c [hunk 1]
```diff
@@ -32,4 +33,47 @@ int netif_get_name(netif_t *iface, char *name)
...
+        // Ask underlying netdev
+        res = dev->driver->get(dev, opt, value, max_len);
+    }
+    return res;
+}
+
+int netif_set_opt(netif_t *iface, netopt_t opt, uint16_t context,
+                  void *value, size_t value_len)
+{
+    return -ENOTSUP;
```
Comment (benpicco) [inline pkg/lwip/contrib/_netif.c:76]:
```suggestion
    (void)context;
    (void)iface;
    (void)opt;
    (void)context;
    (void)value;
    (void)value_len;

    return -ENOTSUP;
```

File: pkg/lwip/contrib/_netif.c [hunk 2]
```diff
@@ -32,4 +33,47 @@ int netif_get_name(netif_t *iface, char *name)
...
+                assert(max_len >= sizeof(ipv6_addr_t));
+                ipv6_addr_t *tgt = value;
+
+                res = 0;
+                for (unsigned i = 0;
+                     ((res + sizeof(ipv6_addr_t)) <= max_len) &&
+                     (i < LWIP_IPV6_NUM_ADDRESSES);
+                     i++) {
+                    if (netif_ip6_addr_state(netif, i) != IP6_ADDR_INVALID) {
+                        memcpy(tgt, &(netif_ip6_addr(netif,i)->addr), sizeof(ipv6_addr_t));
```
Comment (benpicco) [inline pkg/lwip/contrib/_netif.c:55]:
```suggestion
                        memcpy(tgt, &(netif_ip6_addr(netif, i)->addr), sizeof(ipv6_addr_t));
```

**Comment by yarrick** [comment]
Thanks. Squashing right away due to tiny size

**Review by benpicco — APPROVED** [review]

**Comment by miri64** [comment]
Looks like Murdock had some unrelated problems.

File: pkg/lwip/contrib/_netif.c [hunk 3]
```diff
@@ -32,4 +33,53 @@ int netif_get_name(netif_t *iface, char *name)
     return res;
 }
 
+int netif_get_opt(netif_t *iface, netopt_t opt, uint16_t context,
+                  void *value, size_t max_len)
+{
+    lwip_netif_t *lwip_netif = (lwip_netif_t*) iface;
```
Comment (benpicco) [inline pkg/lwip/contrib/_netif.c:39]:
```suggestion
    (void)context;
    lwip_netif_t *lwip_netif = (lwip_netif_t*) iface;
```
still an unused variable left

**Comment by yarrick** [comment]
Seems like something rust-related is broken, also for another PR

**Pull request merged**